admin/assets/javascripts/util-checkbox.js

```diff
@@ -152,7 +152,9 @@ var inputStatus = {
     * @param checkboxParent
     */
   countCheckedCheckboxes: function (checkboxParent) {
-    var el = $((checkboxParent || '.multiple-choice-mtc') + ' > input:checkbox:checked').not('#tickAllCheckboxes')
```

Comment:
@GeorgeChatzigiannis 
all that's needed here is an extra attribute filter to only count visible checked checkbox inputs.

 `var el = $((checkboxParent || '.multiple-choice-mtc') + ' > input:checkbox:checked:visible').not('#tickAllCheckboxes')`
 
Then you don't need to do the adding and removing classes dance - it's cleaner.